R/check_data.R

```diff
@@ -93,3 +93,30 @@ check_company_ticker_mapping <- function(data) {
   }
   return(invisible(data))
 }
+
+#' Check expected missing patterns
+#'
+#' On certain variables missings appear that are not problematic but are
+#' expected. Here it is checked whether those missings adher to expected
+#' patterns.
+#'
+#' @param data Data that expected missings are checked on.
+#'
+#' @return Returns input invisibly.
+check_expected_missings <- function(data) {
+  n_missings_production_plan_company_technology <- sum(is.na(data$production_plan_company_technology))
+  expected_n_missings_production_plan_company_technology <- nrow(data) * (length(unique(data$year)) - length(seq(min(data$year), min(data$year) + time_horizon_lookup)) / length(unique(data$year)))
```

Comment:
@MirijaHa I missed one set of brackets. This line needs to be: `expected_n_missings_production_plan_company_technology <- nrow(data) * ((length(unique(data$year)) - length(seq(min(data$year), min(data$year) + time_horizon_lookup))) / length(unique(data$year)))`